static/src/javascripts/projects/commercial/modules/third-party-tags.js

```diff
@@ -78,10 +79,10 @@ const insertScripts = (
             // TCFv2 mode,
             consentedAdvertisingServices = advertisingServices.filter(
                 script => {
-                    if (
-                        typeof script.sourcepointId !== 'undefined' &&
-                        typeof state.tcfv2.vendorConsents !== 'undefined'
-                    ) {
+                    if (typeof script.sourcepointId !== 'undefined' &&
+                        typeof state.tcfv2.vendorConsents !== 'undefined' &&
+                        typeof state.tcfv2.vendorConsents[script.sourcepointId] !== 'undefined')
```

Comment:
This line can be removed, as a value of `undefined` would filter the script out.

Reply:
Agree for cases which `state.tcfv2.vendorConsents` the extra property check is not needed, but it's needed when the `state.tcfv2.vendorConsents[script.sourcepointId]` is undefined. We want it to go to the `else` in this case as there is no explicit vendorGrant for this sourcepoint id